### PR TITLE
Fix bugs and add improvements according to PE-D

### LINKED ISSUE #146: Unlabelled connection in storage component
Add a label for the arrow coming into storage component

image

### LINKED ISSUE #158: [PE-D][Tester B] Missing Glossary in DG
Glossary is missing in DG. Adding one can be helpful to readers as there are many complicated and non intuitve terms such as "cashflow" in the program.

-------------
Labels: `type.DocumentationBug` `severity.Medium`
original: DaDevChia/ped#4

### LINKED ISSUE #162: [PE-D][Tester E] User Guide - File name is wrong
In Quick Start, The File name is ip.jar, but the actual released file name is tp.jar.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: junhyeong0411/ped#1

### LINKED ISSUE #163: [PE-D][Tester B] Lack of activation bars in most sequence diagrams in DG
Activation bars are not present in most of the sequence diagrams shown in the DG. Could adding them help the reader better understand when the objects are active?

-------------
Labels: `severity.High` `type.DocumentationBug`
original: DaDevChia/ped#6

### LINKED ISSUE #164: [PE-D][Tester E] Cannot download Financial Planner file
Following the User Guide, I tried to download Financial Planner, but I couldn't. Even I typed my email, nothing happen.

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: junhyeong0411/ped#2

### LINKED ISSUE #165: [PE-D][Tester C] Exception Message Thrown
The message thrown for budget set could be better described. I typed budget and the message showed something a user who has no knowledge in programming would not understand. Small bug but just a thought for improvement to give a more informative message in layman term.

Screenshot 2023-11-03 at 5.00.16 PM.png

-------------
Labels: `type.FunctionalityBug` `severity.Low`
original: J0shuaLeong/ped#2

### LINKED ISSUE #169: [PE-D][Tester E] Logs should be hided
Using the program, logs are not hided, so it doesn't follow the expected output for users.

You have to change the log level before complie the jar file.

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: junhyeong0411/ped#3

### LINKED ISSUE #173: [PE-D][Tester F] LOGGER triggered when there is no `data/` folder
**Note from the teaching team:** This bug was reported during the _Part II (Evaluating Documents)_ stage of the PE. **You may reject this bug if it is not related to the quality of documentation.**<hr>

Marked with `severity.HIGH` as this is likely to impact all users upon first use of the application.

-------------
Labels: `type.FunctionalityBug` `severity.High`
original: lctxct/ped#1

### LINKED ISSUE #177: [PE-D][Tester A] Storage file contains budget, even though i didnt add it
**Note from the teaching team:** This bug was reported during the _Part II (Evaluating Documents)_ stage of the PE. **You may reject this bug if it is not related to the quality of documentation.**<hr>

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: 000verflow/ped#4

### LINKED ISSUE #180: [PE-D][Tester F] Incorrect message when budget exceeded
**Note from the teaching team:** This bug was reported during the _Part II (Evaluating Documents)_ stage of the PE. **You may reject this bug if it is not related to the quality of documentation.**<hr>
Steps to reproduce: 
```
add income /a 10 /t salary 
budget set /b 10 
add expense /a 11 /t others 
```

Upon viewing budget with `budget view`, the following error message is received 
```
You have exceeded your current budget by: -1.00
```
where the correct value should be `1.00` (exceed by `positive number`).

-------------
Labels: `type.FunctionalityBug` `severity.VeryLow`
original: lctxct/ped#2

### LINKED ISSUE #184: [PE-D][Tester F] Hard to differentiate between output and input
**Note from the teaching team:** This bug was reported during the _Part II (Evaluating Documents)_ stage of the PE. **You may reject this bug if it is not related to the quality of documentation.**<hr>
At least on Windows Powershell... :( 

Can consider adding some way to differentiate between user-supplied input and output. 

-------------
Labels: `severity.VeryLow` `type.FeatureFlaw`
original: lctxct/ped#5

### LINKED ISSUE #190: [PE-D][Tester D] Text Formatting
**Note from the teaching team:** This bug was reported during the _Part II (Evaluating Documents)_ stage of the PE. **You may reject this bug if it is not related to the quality of documentation.**<hr>
More of a suggestion, but I think it would be a lot easier on the eyes if when a user calls the overview command it is formatted better, as in the current state, it looks just like the rest of the text and is a little hard to read.

-------------
Labels: `type.FeatureFlaw` `severity.VeryLow`
original: Remy9926/ped#5

### PR DESCRIPTION
**Pull request by ryan1604**
Fixes #146
Fixes #158 
Fixes #162 
Fixes #163 
Fixes #164 
Fixes #165 
Fixes #169 
Fixes #173 
Fixes #177 
Fixes #180 
Fixes #184 
Fixes #190

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -88,13 +88,15 @@ API: `Storage.java`
...
-* Saving the data once upon exit (Currently implemented):
+* Option 1 (Currently implemented): Saving the data once upon exit
   * Advantage: Better efficiency and performance of the program.
   * Disadvantage: If the program crashes or exits incorrectly, data will not be saved.
-* Saving the data after every command:
+* Option 2: Saving the data after every command:
   * Advantage: Changes are saved after every command.
   * Disadvantage: Executing command might slow down the program when there is a large amount of data to be saved.
 
+Option 1 is chosen to prioritise the performance of the program.
```
Comment (wwweert123) [inline docs/DeveloperGuide.md:98]: Thanks for clarifying

File: docs/UserGuide.md [hunk 2]
```diff
@@ -51,12 +51,12 @@ you a one-stop interface to access a plethora of features to manage your finance
 
 ## Quick Start
 
-{Give steps to get started quickly}
+{TO BE UPDATED WHEN V2.1 IS COMPLETED!!!}
 
 1. Ensure that you have Java 11 or above installed.
-2. Download the latest version of `Financial Planner` from [here](http://link.to/duke).
+2. Download the latest version of `Financial Planner` from [here](https://github.com/AY2324S1-CS2113-T18-2/tp/releases).
```
Comment (wwweert123) [inline docs/UserGuide.md:57]: Thanks for adding the correct link

File: docs/diagrams/Storage.puml [hunk 3]
```diff
@@ -7,24 +7,29 @@ package Storage as StoragePackage {
...
 Class "<<enumeration>>\nIncomeType" as IncomeType
 Class Ui
+Class Lists
 }
 
-Class FinancialPlanner #FFFFFF
+note "Cashflow refers to income and expense" as N1
+N1 -up-> Cashflow
+
+note "Lists refer to CashflowList, ReminderList, WishList" as N2
```
Comment (wwweert123) [inline docs/diagrams/Storage.puml:21]: Good use of notes

File: src/main/java/seedu/financialplanner/FinancialPlannerLogger.java [hunk 4]
```diff
@@ -17,6 +18,10 @@ public class FinancialPlannerLogger {
      */
     public static void initialise() {
         try {
+            File dir = new File("data");
```
Comment (wwweert123) [inline src/main/java/seedu/financialplanner/FinancialPlannerLogger.java:21]: 👍

File: src/main/java/seedu/financialplanner/utils/Ui.java [hunk 5]
```diff
@@ -47,7 +48,7 @@ public void setScanner(Scanner scanner) {
 
     public void showMessage(String message) {
         assert !message.isEmpty();
-        System.out.println(message);
+        System.out.println(CYAN + message + RESET);
```
Comment (wwweert123) [inline src/main/java/seedu/financialplanner/utils/Ui.java:51]: Nice Cyan

**Review by wwweert123 — APPROVED** [review]
LGTM

File: docs/images/Budget.png
Comment (NeoMinWei) [inline docs/images/Budget.png:1]: add a final return for the execute() so the bar does not continue off the diagram

**Review by NeoMinWei — APPROVED** [review]
LGTM
